src/main/java/cpw/mods/modlauncher/ClassCache.java

```diff
@@ -0,0 +1,99 @@
+package cpw.mods.modlauncher;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ClassCache {
+    static transient final int VERSION = 1;
+    static File classCacheFile, cacheConfigFile, tempClassCacheFile;
+    static boolean validCache = true;
+    static Map<String, byte[]> classCacheToWrite = new ConcurrentHashMap<>();
+
+    static void init(File baseDir)
+    {
+        baseDir.mkdirs();
+        classCacheFile = new File(baseDir + "/cache.jar");
+        cacheConfigFile = new File(baseDir + "/configuration.cfg");
+        tempClassCacheFile = new File(baseDir + "/cache.jar.temp");
+        try
+        {
+            if (tempClassCacheFile.exists())
+            {
+                Files.move(tempClassCacheFile.toPath(), classCacheFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Modlauncher targets Java 8+, how about using [Paths](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html) instead of `File`?